### PR TITLE
Upgrade msvc port to winsock2

### PR DESCRIPTION
**Pull request by Skptak**
Description
-----------
MBedTLS swaps to using WinSock2.h in their files after the V3.3.0 Version.
Upgrade the MSVC port to use WinSock2.h inorder to be compatible with future mbedtls releases.

Test Steps
-----------
Ran tests in FreeRTOS/FreeRTOS

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/kernel-demos.yml [hunk 1]
```diff
@@ -9,8 +9,8 @@ jobs:
       - name: Checkout the FreeRTOS/FreeRTOS Repository
         uses: actions/checkout@v3
         with:
-          ref: main
-          repository: FreeRTOS/FreeRTOS
+          ref: msvcDemoWinsockTwo
```
Comment (aggarg) [inline .github/workflows/kernel-demos.yml:12]: This is not recommended.

File: portable/MSVC-MingW/port.c [hunk 2]
```diff
@@ -29,12 +29,31 @@
 /* Standard includes. */
 #include <stdio.h>
 
+/** Exclude Windows APIs such as Cryptography, DDE, RPC, Shell,
```
Comment (aggarg) [inline portable/MSVC-MingW/port.c:32]: I am not clear on why these changes are needed here. Would you please elaborate?

File: portable/MSVC-MingW/portmacro.h [hunk 3]
```diff
@@ -29,7 +29,21 @@
 #ifndef PORTMACRO_H
 #define PORTMACRO_H
 
+/** Exclude Windows APIs such as Cryptography, DDE, RPC, Shell,
```
Comment (aggarg) [inline portable/MSVC-MingW/portmacro.h:32]: Same as above.

File: .github/workflows/kernel-demos.yml (hunk 1 above)
Comment (Skptak) [reply .github/workflows/kernel-demos.yml:12]: Currently the changes here break the existing MSVC demos in [FreeRTOS/FreeRTOS/Demo/WIN32-MSVC](https://github.com/FreeRTOS/FreeRTOS/tree/main/FreeRTOS/Demo/WIN32-MSVC), and [FreeRTOS/FreeRTOS/Demo/WIN32-MSVC-Static-Allocation-Only](https://github.com/FreeRTOS/FreeRTOS/tree/main/FreeRTOS/Demo/WIN32-MSVC-Static-Allocation-Only).
We'll need to merge a "failing" PR into here, or into FreeRTOS/FreeRTOS. For now I have this pointed to my fork to show the changes to this port work with the changes to the demo projects.

File: portable/MSVC-MingW/port.c (hunk 2 above)
Comment (Skptak) [reply portable/MSVC-MingW/port.c:32]: Sorry, I had written a short explanation of this in the description, but I should have explained more.

In MbedTLS versions after, and including, v3.5.0 they started defining `WIN32_LEAN_AND_MEAN`, and using `winsock2.h` The commit for this can be found [here](https://github.com/Mbed-TLS/mbedtls/commit/c26bd760204b27149a4bb08bf2604cc34d3a85fd), and the blame can be found [here](https://github.com/Mbed-TLS/mbedtls/blame/v3.5.0/library/x509_crt.c#L2697)

As such in order to upgrade the MbedTLS version from 3.2.1 to 3.5.1 in the FreeRTOS/FreeRTOS repo we need to use winsock2.h in the MSVC port.

Does that explain why this is being done?

**Review by sirnish — APPROVED** [review]
